src/main/java/bio/terra/service/auth/iam/sam/SamIam.java

```diff
@@ -166,20 +166,25 @@ private Map<UUID, Set<IamRole>> listAuthorizedResourcesInner(
   }
 
   @Override
-  public boolean hasActions(
+  public List<String> listActions(
       AuthenticatedUserRequest userReq, IamResourceType iamResourceType, String resourceId)
       throws InterruptedException {
     return SamRetry.retry(
-        configurationService, () -> hasActionsInner(userReq, iamResourceType, resourceId));
+        configurationService, () -> listActionsInner(userReq, iamResourceType, resourceId));
   }
 
-  private boolean hasActionsInner(
+  private List<String> listActionsInner(
       AuthenticatedUserRequest userReq, IamResourceType iamResourceType, String resourceId)
       throws ApiException {
     ResourcesApi samResourceApi = samResourcesApi(userReq.getToken());
-    List<String> actionList =
-        samResourceApi.resourceActionsV2(iamResourceType.toString(), resourceId);
-    return (actionList.size() > 0);
+    return samResourceApi.resourceActionsV2(iamResourceType.toString(), resourceId);
+  }
+
+  @Override
+  public boolean hasActions(
```

Comment:
Not for this PR but I've always found this name to be a bit strange.  Like, it implies that you'd be able to pass in a set of actions.  Eventually, we may want to rename to `hasAnyActions`.

Reply:
Heh, I had this reaction too, so I think it's fair to include in this PR!